util/autogen_banner.py

```diff
@@ -0,0 +1,49 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from textwrap import wrap
+
+LICENSE_BANNER = [
+    "Copyright lowRISC contributors.",
+    "Licensed under the Apache License, Version 2.0, see LICENSE for details.",
+    "SPDX-License-Identifier: Apache-2.0",
+]
+
+AUTOGEN_BANNER = [
+    "-------------------------------------------------------------------------",
+    "-------- W A R N I N G: A U T O - G E N E R A T E D   C O D E !! --------",
+    "DO NOT EDIT THIS FILE DIRECTLY. GENERATED WITH THE FOLLOWING COMMAND:",
+    "-------------------------------------------------------------------------",
+]
+
+
+def get_autogen_banner(command: str, comment=None) -> str:
+    """Returns a commented out auto-generated code warning banner.
+
+    command is a fully formatted string representing what command was used
+    to auto-generate the source.
+    comment is the style of comment supported by the file type.
+    """
+    command_lines = wrap(command.strip(), width=len(AUTOGEN_BANNER[0]))
+    banner = AUTOGEN_BANNER[:-1] + command_lines + [AUTOGEN_BANNER[-1]]
```

Comment:
It probably makes sense to, instead of making assumptions about the size of the banner, to include a token in the banner so that you can `.format()` it. E.g. `AUTOGEN_HEADER.format(command=command)`.

(I would also suggest making the constant a string; apply comment would better apply to a string than to a list.)

Reply:
The only assumption made here is that the command (now wrapped into a list of lines) needs to be inserted before the last line. Given that `AUTOGEN_BANNER` is file local constant, it should be ok to make such assumptions. 

The comment needs to be applied to all lines of the text. It is hence, better to keep the constant as a list rather than string. Otherwise we need to split it on newlines and join again.

Reply:
nit: Most autogen messages I have seen lately are as simple as this. The prefix, suffix lines are not as common anymore. This code would look simpler without the line suffix. Leaving as nit.

```cc
// THIS FILE HAS BEEN GENERATED, DO NOT EDIT MANUALLY.
// Command: {command}
```

Reply:
Done.